src/Stratis.Features.FederatedPeg/OpReturnDataReader.cs

```diff
@@ -107,7 +109,7 @@ private string TryConvertValidOpReturnDataToAddress(byte[] data)
             // Attempt to parse the string. Validates the base58 string.
             try
             {
-                var bitcoinAddress = this.network.ToCounterChainNetwork().Parse<BitcoinAddress>(destination);
+                var bitcoinAddress = this.counterChainNetwork.Parse<BitcoinAddress>(destination);
```

Comment:
Why not just inject and INetworkAddressParser implementation for this and copy the behaviour, it will simplify things so Networks don't have to be referencing each other

Reply:
- There are other places we will want to know about the counter-chain network anyway for future optimisations (e.g. block to sync from for federation wallet. when to sync deposits from etc).

- We would have to implement and inject a different instance for each of main, test, regtest - we don't do this anywhere else and it would push logic into the daemons